src/components/WalletSelector/multichain/tezos/TezosAuthenticateWallet.tsx

```diff
@@ -0,0 +1,157 @@
+
+  const { handleLogin } = useAuthActions();
+
+  const createNonce = useCreateNonce();
+  const loginOrRedirectToOnboarding = useLoginOrRedirectToOnboarding();
+
+  const trackSignInAttempt = useTrackSignInAttempt();
+  const trackSignInSuccess = useTrackSignInSuccess();
+  const trackSignInError = useTrackSignInError();
+
+  /**
+   * Auth Pipeline:
+   * 1. Fetch nonce from server with provided wallet address
+   * 2. Sign nonce with wallet (metamask / walletconnect / etc.)
+   * 3a. If wallet exists, log user in
+   * 3b. If wallet is new, sign user up
+   */
+  const attemptAuthentication = useCallback(
+    async (address: string, publicKey: string) => {
+      console.log('Requesting permissions...');
+
+      setPendingState(PROMPT_SIGNATURE);
+      trackSignInAttempt('Tezos');
+
+      const { nonce, user_exists: userExists } = await createNonce(address, 'Tezos');
+
+      const formattedInput: string = ['Tezos Signed Message:', nonce].join(' ');
+
+      // https://tezostaquito.io/docs/signing
+      const bytes = char2Bytes(formattedInput);
+      const payloadBytes = '05' + '01' + '00' + char2Bytes(bytes.length.toString()) + bytes;
+
+      const payload: RequestSignPayloadInput = {
+        signingType: SigningType.MICHELINE,
+        payload: payloadBytes,
+        sourceAddress: address,
+      };
+
+      const { signature } = await beaconClient.requestSignPayload(payload);
+
+      // Get the nonce number
+      const splittedNonceMessage = formattedInput.split(' ');
+      const nonceNumber = splittedNonceMessage[splittedNonceMessage.length - 1];
+
+      const userId = await loginOrRedirectToOnboarding({
+        authMechanism: {
+          mechanism: {
+            eoa: {
+              chainPubKey: {
+                pubKey: publicKey,
+                chain: 'Tezos',
+              },
+              nonce: nonceNumber,
+              signature,
+            },
+          },
+        },
+        userExists,
+      });
+
+      if (userExists && userId) {
+        trackSignInSuccess('Tezos');
+        return await handleLogin(userId, address);
+      }
+    },
+    [
+      trackSignInAttempt,
+      createNonce,
+      loginOrRedirectToOnboarding,
+      trackSignInSuccess,
+      handleLogin,
+      beaconClient,
+    ]
+  );
+
+  useEffect(() => {
+    async function authenticate() {
+      try {
+        const { publicKey, address } = await beaconClient.requestPermissions();
+
+        if (!address || !publicKey) return;
+
+        await attemptAuthentication(address, publicKey);
+      } catch (error) {
+        trackSignInError('Tezos', error);
+        // ignore early access errors
+        if (!isEarlyAccessError(error)) {
+          // capture all others
+          captureException(error);
+        }
+        setError(normalizeError(error));
+      }
+    }
+
+    void authenticate();
+  }, [attemptAuthentication, trackSignInError, beaconClient]);
+
+  if (error) {
+    return (
+      <EthereumError
```

Comment:
nit: this is called `EthereumError`, maybe we could call it `WalletError` or something more generic